### DISCUSSION TITLE
[V3] Bug: Interact With Component by JavaScript

### DISCUSSION THREAD

**Discussion by devajmeireles** [livewire/livewire#5970] [7. Bugs (archived)] [answered]
# Code, Frontend

```php
import './bootstrap';
import swal from 'sweetalert';

import { Livewire, Alpine } from '../../vendor/livewire/livewire/dist/livewire.esm';

Alpine.data('test', (test) => ({
    test: test,
    init() {
        console.log(this.test)
    },
    testing() {
        Livewire.first().$dispatch('test'); // 👈  it works...
        
        Livewire.first().test(); // 👈  removing attribute #[On] and trying to call it directly, doesn't work...
    }
}));

Livewire.start()
```

# Code, Backend

```php
class Create extends Component
{
    //...

    #[On('test')]
    public function test()
    {
        $this->js("alert('OK');");
    }
}
```

# Using

The `test()` method of Alpine Data is called on a Blade page that is part of a Livewire fullpage component. I'm doing something wrong?

![CleanShot 2023-07-25 at 5 37 25](https://github.com/livewire/livewire/assets/60591772/2adc887e-f054-41cd-8ff0-58a6614e3a63)

# Error

![CleanShot 2023-07-25 at 5 38 13](https://github.com/livewire/livewire/assets/60591772/d1df065c-97b6-4513-b666-c92c757ad0ee)

**Comment by devajmeireles** [comment]
I think I found another bug. 

It works 👇

```js
Alpine.data('test', () => ({
   testing() {
        let components = Livewire.all();

        for (let component in components) {
            console.log(components[component].id);
        }
    }
}));
```

It doesn't work 👇 

```js
Alpine.data('test', () => ({
   testing() {
        let component = Livewire.first();
        
        console.log(component.id);
    }
}));
```

The log of this above code is 👇 

![CleanShot 2023-07-25 at 6 56 44](https://github.com/livewire/livewire/assets/60591772/fe3be31d-f53a-4be3-af8b-6f9d635ba300)

**Comment by joshhanley** [comment, marked as answer]
@devajmeireles `Livewire.first()` returns the `$wire` proxy for that component, so if you want the instance, do `Livewire.first().__instance`.

Hope this helps!
